Sources/Benchmark/Benchmark.swift

```diff
@@ -14,23 +14,31 @@
 
 public protocol AnyBenchmark {
     var name: String { get }
+    var settings: [BenchmarkSetting] { get }
     func run()
 }
 
 internal class ClosureBenchmark: AnyBenchmark {
     let name: String
+    let settings: [BenchmarkSetting]
     let closure: () -> Void
 
-    init(_ name: String, _ closure: @escaping () -> Void) {
+    init(_ name: String, settings: [BenchmarkSetting], using f: @escaping () -> Void) {
         self.name = name
-        self.closure = closure
+        self.settings = settings
+        self.closure = f
     }
 
     func run() {
         return self.closure()
     }
 }
 
-public func benchmark(_ name: String, f: @escaping () -> Void) {
-    defaultBenchmarkSuite.benchmark(name, f)
+public func benchmark(_ name: String, using f: @escaping () -> Void) {
+    defaultBenchmarkSuite.benchmark(name, using: f)
+}
+
+public func benchmark(_ name: String, settings: BenchmarkSetting..., using f: @escaping () -> Void)
```

Comment:
Nice use of the variadic!

```suggestion
public func benchmark(_ name: String, settings: BenchmarkSetting..., function: @escaping () -> Void)
```

I think `function` might be a better name than `using f`. WDYT?

Reply:
Sounds good, I changed as suggested.